### DISCUSSION TITLE
Question: Image to Prompt Possible with this SDK?

### DISCUSSION THREAD

**Discussion by bin2ai** [Stability-AI/stability-sdk#141] [Q&A]
There is a function inside the [client.py](https://github.com/Stability-AI/stability-sdk/blob/main/src/stability_sdk/client.py) script called image_to_prompt which returns type generation.prompt which appears to be a byte string.

Has anyone had any luck converting an image to prompt (ASCII) using this SDK?

**Comment by shishimaru** [comment]
@bin2ai 
Hi,

I also found the internal function and there is a sample program to get as ASCII string.
https://colab.research.google.com/github/pharmapsychotic/clip-interrogator/blob/main/clip_interrogator.ipynb#scrollTo=xpPKQR40qvz2

The function you want would be below.


```
def image_to_prompt(image, mode):
    ci.config.chunk_size = 2048 if ci.config.clip_model_name == "ViT-L-14/openai" else 1024
    ci.config.flavor_intermediate_count = 2048 if ci.config.clip_model_name == "ViT-L-14/openai" else 1024
    image = image.convert('RGB')
    if mode == 'best':
        return ci.interrogate(image)
    elif mode == 'classic':
        return ci.interrogate_classic(image)
    elif mode == 'fast':
        return ci.interrogate_fast(image)
    elif mode == 'negative':
        return ci.interrogate_negative(image)
```

On the Web UI, the one was shown.


![image](https://user-images.githubusercontent.com/570090/233577442-a911975a-b7ba-4917-82f8-75b2fc8010e0.png)
